src/app/services/authentication/authentication.service.ts

```diff
@@ -22,12 +23,82 @@ export class AuthenticationService implements OnDestroy {
   private authenticationMethod: AuthenticationMethod;
   private _accessLevel: AccessLevel = AccessLevel.Restricted;
 
+  static getAccessLevelFromRoles(roles: string[]): AccessLevel {
```

Comment:
I am not sure about this part, it seems that no matter what roles I choose and which user, I always have the `Full` access. What should I do to make it `Restricted`?

Reply:
If you have only `SUMMARY` or `COUNTS_WITH_THRESHOLD` access to all studies that you have access to.

Reply:
(see tests). But the code has become a bit messy, because the test script also wants to support es2015, and then you cannot use `some`, `every` or `for let roles of rolesPerStudy.values()`.

Reply:
Basically: add role `<STUDY_ID>|SUMMARY` to your user in Keycloak and make sure you're not admin.